src/models/DefaultData.js

```diff
@@ -153,51 +109,27 @@ class DefaultData {
    * @return {Boolean} Indicate if component is removed or not.
    */
   removeComponentById(id) {
-    const isRemoved = this.__removeComponentById(this.components, id);
-
-    this.__removeRefAttributeById(this.components, id);
-
-    return isRemoved;
-  }
+    let isRemoved = false;
 
-  /**
-   * Remove components by id.
-   * If not present in the provided components, look on their children.
-   * @param {Component[]} components - Components to search the id.
-   * @param {String} id - Id to remove.
-   * @return {Boolean} Indicate if the component is removed.
-   * @private
-   */
-  __removeComponentById(components, id) {
-    // Remove sonar code smell, because its more optimized with a for loop.
-    for (let index = 0; index < components.length; index += 1) { // NOSONAR
-      if (components[index].id === id) {
-        components.splice(index, 1);
+    this.getChildren(id).forEach((component) => {
+      this.removeComponentById(component.id);
+      isRemoved = true;
```

Comment:
Probably should not be set here since the component has not been removed yet.

Reply:
Ok I will rework on the return of this method to return unused file :)